### PR TITLE
Video - Bail out if viewer is not visible

### PR DESCRIPTION
**Pull request by aghassemi**

### CODE REVIEW

File: src/service/video-manager-impl.js [hunk 1]
```diff
@@ -132,9 +133,12 @@ class VideoEntry {
     /** @private {boolean} */
     this.userInteracted_ = false;
 
-    /** @const @private {!../service/vsync-impl.Vsync} */
+    /** @private @const {!../service/vsync-impl.Vsync} */
     this.vsync_ = vsyncFor(win);
 
+    /** @private @const {!../service/viewer-impl.Viewer} */
+    this.viewer_ = viewerFor(win);
```
Comment (cramforce) [inline src/service/video-manager-impl.js]: I'd consider not using an instance var for rarely used services.

Comment (aghassemi) [inline src/service/video-manager-impl.js]: `viewer.isVisible()` can be called several times per video, might be worth caching a reference here.

**Comment by cramforce** [comment]
The call is super fast. Not worth caching except for stuff like animations
where you call all the time.

**Comment by aghassemi** [comment]
Done

**Comment by aghassemi** [comment]
ugh, still can't merge. Can you merge this @cramforce ?

**Pull request merged**